jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialog.kt

```diff
@@ -71,7 +79,19 @@ class RunCommandDialog(private val project: Project, private val container: Cont
     override fun doOKAction() {
         super.doOKAction()
         commandsEnteredPreviously.add(command)
-        runCommand()
+        val task = tasks.selected() ?: throw IllegalStateException("Task not Selected")
+        launch {
+            try {
+                EcsExecUtils.checkRequiredPermissions(project, container.service.clusterArn(), task)
```

Comment:
This should be best effort, if they lack the ability to check the permissions, then we should not prevent them from running ecs exec, we just need to tell them it may fail